Description: makes generate-identity.sh update /etc/hosts file
Author: Vasily Smirnov <vasilii.smirnov@globallogic.com>
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
Index: pelion-edge-amd64/usr/bin/generate-identity.sh
===================================================================
--- pelion-edge-amd64.orig/usr/bin/generate-identity.sh
+++ pelion-edge-amd64/usr/bin/generate-identity.sh
@@ -57,6 +57,12 @@ execute () {
             if [ -f ${IDENTITY_DIR}/identity.json ] ; then
                 cp ${IDENTITY_DIR}/identity.json ${IDENTITY_DIR}/identity_original.json
             fi
+
+            if ! grep -q "$internalid" /etc/hosts; then
+                sed -i '/Pelion Edge/d' /etc/hosts && \
+                echo "127.1.2.7 $internalid # Pelion Edge" >> /etc/hosts
+            fi 2>/dev/null
+
             IFS='.' read -ra ADDR <<< "$lwm2mserveruri"
             /usr/lib/pelion/developer_identity/create-dev-identity.sh\
                 -d \
